src/test/java/api/loans/LoanAPITests.java

```diff
@@ -1497,112 +1473,74 @@ public void canSearchByUserId()
-    List<JsonObject> secondPageLoans = getLoans(secondPage);
 
     assertThat(firstPageLoans.size(), is(0));
     assertThat(firstPage.getInteger("totalRecords"), is(0));
-
-    assertThat(secondPageLoans.size(), is(0));
-    assertThat(secondPage.getInteger("totalRecords"), is(0));
   }
 
   @Test
   public void canFilterByLoanStatus()
     throws MalformedURLException,
     InterruptedException,
     ExecutionException,
-    TimeoutException,
-    UnsupportedEncodingException {
+    TimeoutException {
 
     IndividualResource user = usersFixture.charlotte();
     UUID userId = user.getId();
 
-    String queryTemplate = "userId=\"%s\" and status.name=\"%s\"";
-
-    loansClient.create(new LoanBuilder()
+    loansFixture.createLoan(new LoanBuilder()
       .withUserId(userId)
       .withStatus("Open")
       .withItem(itemsFixture.basedUponSmallAngryPlanet())
       .withRandomPastLoanDate());
 
-    loansClient.create(new LoanBuilder()
+    loansFixture.createLoan(new LoanBuilder()
       .withUserId(userId)
       .withStatus("Open")
       .withItem(itemsFixture.basedUponNod())
       .withRandomPastLoanDate());
 
-    loansClient.create(new LoanBuilder()
+    loansFixture.createLoan(new LoanBuilder()
       .withUserId(userId)
       .withItem(
         itemsFixture.basedUponNod())
       .withStatus("Closed")
       .withRandomPastLoanDate());
 
-    loansClient.create(new LoanBuilder()
+    loansFixture.createLoan(new LoanBuilder()
       .withUserId(userId)
       .withStatus("Closed")
       .withItem(itemsFixture.basedUponTemeraire())
       .withRandomPastLoanDate());
 
-    loansClient.create(new LoanBuilder()
+    loansFixture.createLoan(new LoanBuilder()
       .withUserId(userId)
       .withStatus("Closed")
       .withItem(itemsFixture.basedUponUprooted())
       .withRandomPastLoanDate());
 
-    loansClient.create(new LoanBuilder()
+    loansFixture.createLoan(new LoanBuilder()
       .withUserId(userId)
       .withStatus("Closed")
       .withItem(itemsFixture.basedUponInterestingTimes())
       .withRandomPastLoanDate());
 
-    CompletableFuture<Response> openSearchComppleted = new CompletableFuture<>();
-    CompletableFuture<Response> closedSearchCompleted = new CompletableFuture<>();
-
-    client.get(loansUrl(),
-      "query=" + URLEncoder.encode(String.format(queryTemplate, userId, "Open"), "UTF-8"),
-      ResponseHandler.json(openSearchComppleted));
-
-    client.get(loansUrl(),
-      "query=" + URLEncoder.encode(String.format(queryTemplate, userId, "Closed"), "UTF-8"),
-      ResponseHandler.json(closedSearchCompleted));
+    String queryTemplate = "userId=\"%s\" and status.name=\"%s\"";
 
-    Response openLoansResponse = openSearchComppleted.get(5, TimeUnit.SECONDS);
-    Response closedLoansResponse = closedSearchCompleted.get(5, TimeUnit.SECONDS);
+    Response openLoansResponse = loansFixture.getLoans(String.format(queryTemplate, userId, "Open"));
```

Comment:
Does it make sense to add this version of `getLoans` method to the `loansFixture`? 

i.e. `getLoansByQueryTemplate(String query, Object ... args)`

Reply:
Good idea. I've added some strongly typed support for this as well.